WordPress/gutenberg
# Update: Move pattern actions to the editor package.

Similar to https://github.com/WordPress/gutenberg/pull/60395, and https://github.com/WordPress/gutenberg/pull/60092.

The actions related to patterns will be moved to the editor package in this PR, which will enable their use in the edit post package. This includes the ability to export a pattern as JSON, rename, delete, and perform other functions. However, we have not yet moved the "duplicateTemplatePartAction" action as it requires the movement of certain components to the editor package as well. These components will be moved in a separate pull request.

The resetTemplatePart action and resetTemplateAction were joined together as the template action already mostly supported template parts
The delete template and template pattern should be joined together in a follow-up PR.

## Testing Instructions
Open the patterns in dataviews, try all of its actions and verify they work as expected.
